### ISSUE TITLE
Marshal.load returns escaped non-ascii characters in symbols 

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1530]
I know, a bit weird to use non-ascii characters in symbols but MRI returns an expected result.

MRI (2.5.3):

```
irb(main):001:0> a = [:â]
=> [:â]
irb(main):002:0> Marshal.load(Marshal.dump(a))
=> [:â]
```

TruffleRuby (1.0.0-rc10):

```
irb(main):001:0> a = [:â]
=> [:"â"]
irb(main):002:0> Marshal.load(Marshal.dump(a))
=> [:"\xC3\xA2"]
```

**Comment by eregon** [comment]
This bug is inherited from Rubinius's [marshal.rb](https://github.com/rubinius/rubinius/blob/387c488798d8b911f160d36278710474e526d73f/core/marshal.rb#L152). Should be relatively easy to fix.

**Comment by eregon** [comment]
Fixed in 2f4ba368bbf2a541f538003e4fbb353510e293a5 and added specs for it as well as for serializing objects with multibyte instance variable names. It will be in the next release.

**Comment by eregon** [comment]
@deepj This is fixed in the new 1.0.0 RC 11 release, and verifying from IRB looks good. Please comment here if you're still seeing a problem.
